### ISSUE TITLE
Fix recurring RangeError:Maximum call stack size exceeded when using harp compile

### ISSUE THREAD

**Issue by naomiblack** [angular/angular.io#982]
This seems to be an issue with `harp compile` per https://github.com/sintaxi/harp/issues/538

```
[01:59:47] /Users/naomib/github/angular.io/node_modules/harp/node_modules/terraform/lib/helpers/raw.js:423
  filePath = filePath.replace(/^\/|\/$/g, '')
                      ^

RangeError: Maximum call stack size exceeded
  at String.replace (native)
  at Object.exports.shouldIgnore (/Users/naomib/github/angular.io/node_modules/harp/node_modules/terraform/lib/helpers/raw.js:423:23)
  at Object.exports.(anonymous function) [as shouldIgnore] (/Users/naomib/github/angular.io/node_modules/harp/node_modules/terraform/lib/helpers/memoized.js:21:27)
```

**Comment by jvandemo** [comment]
@naomiblack &mdash; What version of harp are you running and what command are your running? Thanks!

**Comment by kwalrath** [comment]
I got this error yesterday with a brand new, clean angular.io repo, and I suspect my setup is very close to Naomi's.

I'm using the command `gulp check-deploy`. I'm not sure which version of harp I'm using. There's one apparently installed by homebrew, but another copy under angular.io in `node_modules/harp/bin`. I'd _hope_ that the gulp command would use the latter.

```
bin (master)$ harp --version
/.../homebrew/lib/node_modules/harp/node_modules/terraform/node_modules/node-sass/lib/extensions.js:146
    throw new Error('`libsass` bindings not found. Try reinstalling `node-sass`?');
...
bin (master)$ pwd
/Users/.../angular.io-CLEAN/node_modules/harp/bin
bin (master)$ ./harp --version
0.20.1
```

**Comment by jvandemo** [comment]
@kwalrath &mdash; Thank you, I have been able to reproduce the problem and narrow down the issue.

I have updated https://github.com/sintaxi/harp/issues/538 and am waiting for @sintaxi's input before creating a PR with a possible fix.

**Comment by kwalrath** [comment]
Good to hear! I'm unable to deploy angular.io (at least, using my normal workflow) until the fix is in.

**Comment by jvandemo** [comment]
@kwalrath &mdash; I have created a gist for you in case you want to deploy angular.io before there is a fix.

Replace `node_modules/harp/lib/index.js` with the code from [this gist](https://gist.github.com/jvandemo/cb66259aa0b0b286ed2a472a71170454).

It uses `async.each` on lines https://gist.github.com/jvandemo/cb66259aa0b0b286ed2a472a71170454#file-index-js-L250 and https://gist.github.com/jvandemo/cb66259aa0b0b286ed2a472a71170454#file-index-js-L254 instead of `async.eachLimit` in the original version.

That resolves the problem here locally for me allowing me to compile the angular.io website.

Hope that helps (until a permanent solution is in)!

**Comment by kwalrath** [comment]
I was able to deploy angular.io. Many thanks, @jvandemo!

**Comment by jvandemo** [comment]
@kwalrath &mdash; Awesome, thank you, glad to hear it worked :+1:

**Comment by sintaxi** [comment]
Great job! eachLimit should make builds faster as it compiles multiple files at once. Since the bottle neck here is probably the CPU and not I/O it probably does not make a big difference. Ill happily accept a pull request for this. Thanks!

**Comment by jvandemo** [comment]
@sintaxi &mdash; I will create a PR immediately so this can be fixed ASAP. Thanks!

**Comment by sintaxi** [comment]
:+1: right on!

**Comment by jvandemo** [comment]
@sintaxi &mdash; Pull request has been created: https://github.com/sintaxi/harp/pull/548.

All tests pass successfully here. Thanks for your quick follow-up!

**Comment by naomiblack** [comment]
Thank you!!!

**Comment by jvandemo** [comment]
Welcome, glad I could help :+1:

**Comment by naomiblack** [comment]
closed in 0fdceec96342aef96f63cbb8dfb2696365ba7488 which updates to the latest version of harp, with fix from @jvandemo and @sintaxi. THANK YOU!
